lib/src/solver/version_solver.dart

```diff
@@ -350,9 +330,10 @@ class VersionSolver {
     /// Prefer packages with as few remaining versions as possible, so that if a
     /// conflict is necessary it's forced quickly.
     var package = await minByAsync(unsatisfied, (package) async {
-      // If we're forced to use the latest version of a package, it effectively
-      // only has one version to choose from.
-      if (_useLatest.contains(package.name)) return 1;
+      /// Artifically set the packages in [_solveFirst] to have a low
+      /// version count to solve them first. This enables them to not be
+      /// artificially constrained by other packages.
+      if (_solveFirst.contains(package.name)) return 1;
```

Comment:
1 is not low enough that also what all the locked packages has... Zero is probably not good, because packages without versions are good at proving a conflict..

May I suggest 0.5?

Reply:
Even better might be to prioritize solveFirst packages internally by least versions..

Reply:
do you think something like:
```
        return math.max(
            0.5,
            await _packageLister(package).countVersions(package.constraint) *
                0.005);
```
would work better? the `solveFirst` packages will be prioritized internally by the least versions, unless they have more than a hundred versions, in which case we still give them priority over the other packages.

Reply:
I would just do something like:
```dart
// ... explain this logic...
var candidates = unsatisfied.where((p) => _solveFirst.contains(p.name));
if (candidates.isEmpty) {
  candidates = unsatisfied;
}
var package = await minByAsync(candidates, (p) => _packageLister(p).countVersions(p.constraint));
```